src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -232,28 +244,34 @@ public async Task<bool> WaitFor100ContinueAsync(CancellationToken cancellationTo
                 return sendRequestContent;
             }
 
-            private void CheckForCompletion()
+            private void Reset()
             {
                 Debug.Assert(Monitor.IsEntered(SyncObject));
-                Debug.Assert(_requestCompletionState != StreamCompletionState.InProgress || _responseCompletionState != StreamCompletionState.InProgress,
-                    $"CheckForCompletion called but neither request nor response is completed");
-
-                if (_requestCompletionState == StreamCompletionState.InProgress || _responseCompletionState == StreamCompletionState.InProgress)
+                Debug.Assert(_requestCompletionState != StreamCompletionState.InProgress);
+                Debug.Assert(_responseCompletionState != StreamCompletionState.InProgress);
+                Debug.Assert(_requestCompletionState == StreamCompletionState.Failed || _responseCompletionState == StreamCompletionState.Failed,
+                    "Reset called but neither request nor response is failed");
```

Comment:
Just to confirm, another way of stating these three asserts is:
```C#
Debug.Assert(
    (_requestCompletionState == StreamCompletionState.Failed && _responseCompletionState == StreamCompletionState.Completed) ||
    (_requestCompletionState == StreamCompletionState.Completed && _responseCompletionState == StreamCompletionState.Failed));
```
right?

Reply:
Oh, no, I guess they could both be failed.  Ok.

Reply:
Yes and yes. Both must be Completed or Failed. At least one must be Failed.